python_modules/dagster/dagster/core/asset_defs/assets_job.py

```diff
@@ -233,32 +234,36 @@ def build_root_manager(
     source_asset_io_manager_keys = {
         source_asset.io_manager_key for source_asset in source_assets_by_key.values()
     }
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore", category=ExperimentalWarning)
+
+        @root_input_manager(required_resource_keys=source_asset_io_manager_keys)
+        def _root_manager(input_context: InputContext) -> Any:
+            source_asset_key = cast(AssetKey, input_context.asset_key)
+            source_asset = source_assets_by_key[source_asset_key]
+            with warnings.catch_warnings():
```

Comment:
we need to have both of these contexts because `_root_manager` is directly called during execution